Revit_Engine/Query/RevitDefiningLine.cs

```diff
@@ -0,0 +1,80 @@
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Adapters.Revit.Elements;
+using BH.oM.Adapters.Revit.Generic;
+using BH.oM.Base;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Xml.Linq;
+using BH.Engine;
+using BH.oM.Geometry;
+using BH.oM.Physical.Elements;
+using BH.Engine.Environment;
+using BH.oM.Environment.Fragments;
+using System.Net.NetworkInformation;
+using BH.Engine.Physical;
+using System.Runtime.Remoting.Messaging;
+
+namespace BH.Engine.Adapters.Revit
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Returns the in Revit defining location line of an IFramingElement, based on its z Justification Revit parameter.")]
+        [Input("element", "The IFramingElement to query the in Revit defining location line of.")]
+        [Output("curve", "The in Revit geometry defining location line of the element.")]
+        public static ICurve RevitDefiningLine(this IFramingElement element)
+        {
+            string zJustification = element.GetRevitParameterValue("z Justification").ToString();
+
+            zJustification = zJustification.ToLower();
+
+            ICurve definingLine = null;
+
+            if (zJustification == "top")
+            {
+                definingLine = element.TopCentreline();
+                return definingLine;
+            }
+            else if (zJustification == "bottom")
+            {
+                definingLine = element.BottomCentreline();
+                return definingLine;
+            }
+            else if (zJustification == "origin" || zJustification == "center")
+            {
+                return element.Location;
+            }
```

Comment:
I just realised this variable isn't really needed because we're not doing any subsequent analysis of the line, just returning it from the element. So we could remove this, and simplify the code to look like this:

```suggestion
            if (zJustification == "top")
                return element.TopCentreline();
            else if (zJustification == "bottom")
                return element.BottomCentreline();
            else if (zJustification == "origin" || zJustification == "center")
                return element.Location;
```

What do you think? This is just a suggestion so making it a PR comment rather than a requested change :smile:

Reply:
Yes agree!